vendor/k8s.io/kubernetes/pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -510,7 +510,7 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 		if containerStatus == nil || containerStatus.State != kubecontainer.ContainerStateRunning {
 			if kubecontainer.ShouldContainerBeRestarted(&container, pod, podStatus) {
 				message := fmt.Sprintf("Container %+v is dead, but RestartPolicy says that we should restart it.", container)
```

Comment:
While I realize that you are not responsible for this behavior, line 512 is an example of time spent formatting messages only to be discarded when the message is not logged on line 513 because the level of the message is not high enough.

This might be worth investigating in general.

Reply:
Filed https://github.com/kubernetes/kubernetes/issues/79071.